src/main/java/uk/gov/hmcts/darts/common/repository/HearingRepository.java

```diff
@@ -39,6 +39,14 @@ public interface HearingRepository extends JpaRepository<HearingEntity, Integer>
     )
     List<Integer> findHearingIdsByEventId(Integer eventId);
 
+    @Query("""
+        SELECT h FROM HearingEntity h
+        JOIN h.mediaList media
+        WHERE media.id = :mediaId
+        """
+    )
+    Optional<List<HearingEntity>> findHearingIdsByMediaId(Integer mediaId);
```

Comment:
A non-null result should always be returned as the collection would simply be empty if the query doesn't match any records. So I don't think the Optional wrapper adds any value/safety? 

See [Best Practices](https://tools.hmcts.net/confluence/pages/viewpage.action?pageId=1694706043#BestPractices(Backend)-RepositoryMethodsShouldReturnOptional<>ForAllNon-CollectionReturnTypes) for more details.